File: C++/2501. Longest Square Streak in an Array.cpp
class Solution {
public:
    int longestSquareStreak(vector<int>& nums) {
        set<double>st;
        const int num=(1e5);
        for(auto x:nums){
            st.insert(x);
        }
        vector<int>temp;
        for(auto x:st){
            temp.push_back(x);
        }
        int ans=0;
        for(auto x:st){
            double val=x;
            val=sqrt(val);
            int c=0;
            while(st.find(val)!=st.end()){
                val=sqrt(val);
                c++;
            }
            ans=max(ans,c);
        }
        return ans==0?-1:ans+1;
        
    }
};

File: JAVA/2501. Longest Square Streak in an Array.java
class Solution {
    public int longestSquareStreak(int[] nums) {
     Set<Integer> st = new HashSet<>();
     for(int i=0;i<nums.length;i++)
     {
        st.add(nums[i]);
     }
     int count=0;
     for(int i=0;i<nums.length;i++)
     {
        int num = nums[i];
        int ct = 0;
        while(st.contains(num))
        {
            ct++;
            if(num>Integer.MAX_VALUE/num)
            break;
            num *= num;
        }
        count = Math.max(count,ct);
     }   
     if(count<=1)
     return -1;
     return count;
    }
}
